sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -557,7 +558,10 @@ private[hive] class HiveClientImpl(
   override def getFunctionOption(
       db: String,
       name: String): Option[CatalogFunction] = withHiveState {
-    Option(client.getFunction(db, name)).map(fromHiveFunction)
+    client.getFunction(db, name)
+    Try {
+      Option(client.getFunction(db, name)).map(fromHiveFunction)
```

Comment:
also we shouldn't wrap this around a `Try`, which would swallow exceptions. This should return `None` only if the function doesn't exist, not when it throws exceptions.